flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -382,84 +392,53 @@ class CodeGenerator(
          |  }""".stripMargin
     }
 
-    def genSetForwardedFields(
-        forwardMapping: Array[(Int, Int)]): String = {
+    def genSetForwardedFields(forwardMapping: Array[Int]): String = {
 
       val sig: String =
         j"""
            |  public void setForwardedFields(
            |    org.apache.flink.types.Row input,
-           |    org.apache.flink.types.Row accs,
            |    org.apache.flink.types.Row output)
            |    """.stripMargin
 
       val forward: String = {
-        for (i <- forwardMapping.indices) yield
-          j"""
-             |    output.setField(
-             |      ${forwardMapping(i)._1},
-             |      input.getField(${forwardMapping(i)._2}));"""
-            .stripMargin
-      }.mkString("\n")
-
-      var copyKeys: String = ""
-      if (gkeyOutMapping != null) {
-        copyKeys = {
-          for ((out, in) <- gkeyOutMapping) yield
+        for (i <- forwardMapping.indices if forwardMapping(i) >= 0) yield
```

Comment:
Yes. The forwardMapping array specifies for the which fields should be copied to the first fields (positions 0 to `forwardMapping.size - 1` fields of the output row. For grouping sets, it can happen that we only need to set position `1` but not position `0`. I solved this by providing a `forwardMapping` array as `(-1, 0)`, which indicates that pos `0` of the output row is not set (hence skipped by the check) and pos `1` is set with the `0`th field of the input row.